From a39101060cdf2ee239833106fb3bdf9585f858aa Mon Sep 17 00:00:00 2001
From: Tom de Vries <tdevries@suse.de>
Date: Wed, 7 Jun 2023 11:36:19 +0200
Subject: [PATCH] [gdb/testsuite] Handle output after prompt in
 gdb.threads/step-N-all-progress.exp

Using "taskset -c 0" I run into this timeout:
...
(gdb) PASS: gdb.threads/step-N-all-progress.exp: non-stop=on: \
  target-non-stop=on: continue to breakpoint: break here
next 3^M
[New Thread 0x7ffff7dbd6c0 (LWP 10202)]^M
50        return 0;^M
(gdb) [Thread 0x7ffff7dbd6c0 (LWP 10202) exited]^M
FAIL: gdb.threads/step-N-all-progress.exp: non-stop=on: target-non-stop=on: \
  next 3 (timeout)
...

The problem is that this test:
...
    gdb_test "next 3" "return 0;"
...
expects no output after the prompt.

Fix this by using -no-prompt-anchor.

Tested on x86_64-linux.
---
 gdb/testsuite/gdb.threads/step-N-all-progress.exp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdb/testsuite/gdb.threads/step-N-all-progress.exp b/gdb/testsuite/gdb.threads/step-N-all-progress.exp
index 98cc8423f34..485028d34e7 100644
--- a/gdb/testsuite/gdb.threads/step-N-all-progress.exp
+++ b/gdb/testsuite/gdb.threads/step-N-all-progress.exp
@@ -44,7 +44,7 @@ proc test {non-stop target-non-stop} {
 
     gdb_continue_to_breakpoint "break here"
 
-    gdb_test "next 3" "return 0;"
+    gdb_test -no-prompt-anchor "next 3" "return 0;"
 }
 
 foreach_with_prefix non-stop {off on} {

base-commit: 7f7fcd7031430953f41b284069d1ed0cf3c8734a
prerequisite-patch-id: b0bf6c343c4113f430e2dce98279078433692131
prerequisite-patch-id: 20ed3972a694ff7b7eebd0626be31f8694c62afe
prerequisite-patch-id: 2d7b3990c49a8ddc0e4ebc5c328b8cfcd841ec93
prerequisite-patch-id: 201c63297f725b3796a241d804df62e02b4a8abc
-- 
2.35.3

